.golangci.yml

```diff
@@ -47,3 +53,9 @@ linters:
     - misspell    # Finds commonly misspelled English words in comments
     - gofmt       # Gofmt checks whether code was gofmt-ed
     - golint      # Golint differs from gofmt. Gofmt reformats Go source code, whereas golint prints out style mistakes
+    - errcheck    # errcheck is a program for checking for unchecked errors in go programs.
+
+linters-settings:
+  errcheck:
+    # Disable warnings for `fmt` and `log` packages.
+    ignore: fmt:.*,github.com/DataDog/datadog-agent/pkg/util/log:.*
```

Comment:
Given the huge number of `http.ResponseWriter.write(...)` requiring the `//nolint:errcheck` would it be possible to exclude  it ? (possible errors seem relatively safe to ignore https://golang.org/pkg/net/http/#pkg-variables but I may be wrong about that)

Reply:
I tried to ignore `http.ResponseWriter.Write` a couple of days ago but because of https://github.com/golangci/golangci-lint/issues/656, it seems I can only ignore all the `Write` methods from http package (and not the specific method http.ResponseWriter.Write). 
Upgrading golangci-lint requires extra work.

Reply:
I agree with @prognant, I'm not a fan of all the `//nolint:errcheck`. We might want to ignore all `Write()` call from the http package at this point.

As a more general review, we ignore so many warning in this PR that I wonder if it's worth it. I feel like it's increasing the code complexity and we're not gaining a lot. Like all call to `json` or `yaml` should actually be fixed instead of being ignore. WDYT ?

I'm afraid people are not actually going to check their error but just pollute the code base with `//nolint:errcheck`. At this point we will fall back to where we are now: relying on review instead of the linter. That's why I didn't enabled it in the first PR for golangci: way to many warning we might not want to fix.

Reply:
my 2 cents:
* I guess it makes sense for the linter to ignore all `Write` calls from the http package. Let's add a fixme comment linking to the golanci issue, we might be able to make the ignore rule more specific when we upgrade golangci
* I prefer explicitly calling out all instances of unhandled errors with `//nolint:errcheck` than having errors "silently" ignored in the code (by "silent" here, I mean that the calling code doesn't even acknowledge that the call can return an error)
* merging this with a bunch of `//nolint:errcheck` is simpler to review and merge, and already provides value for future code we'll add to the codebase. But yes please create tasks in the backlog about cleaning up all these `//nolint:errcheck` (one card per team owning related packages)